### PR TITLE
Fix #5617 - respect message and reconnect parameters when shutting down immediately

### LINKED ISSUE #5617: minetest.request_shutdown("whatever", true) does not cause the client to display a reconnect button.
Also, it appears that the message specified in "whatever" is not sent to the client, either, causing a generic shutdown message to be displayed instead of the custom message.

### PR DESCRIPTION
**Pull request by orwell96**
This fixes issue #5617.
This fix is UNTESTED, please test.
EDIT: it works for BluebirdGreycoat.

### CODE REVIEW

**Comment by BluebirdGreycoat** [comment]
I've tested. It works for me.

File: src/server.cpp [hunk 1]
```diff
@@ -3479,9 +3479,15 @@ v3f Server::findSpawnPos()
...
 void Server::requestShutdown(const std::string &msg, bool reconnect, float delay)
 {
+	m_shutdown_timer = delay;
+	m_shutdown_msg = msg;
+	m_shutdown_ask_reconnect = reconnect;
+
 	if (delay == 0.0f) {
 	// No delay, shutdown immediately
 		m_shutdown_requested = true;
+		// only print to the infostream, a chat message saying "Server Shutting Down" is sent when the server destructs.
```
Comment (SmallJoker) [inline src/server.cpp]: Line to long.

**Comment by orwell96** [comment]
Changed!

**Comment by nerzhul** [comment]
merged thanks

**Pull request merged**